connect/transforms/src/test/java/org/apache/kafka/connect/transforms/InsertHeaderTest.java

```diff
@@ -0,0 +1,300 @@
+        props.put("header", "AAA");
+        props.put("literal.value", "50000");
+
+        xform.configure(props);
+
+        final SourceRecord record = new SourceRecord(null, null, "test",
+            0, null, null, null, null, null, null);
+        final SourceRecord transformedRecord = xform.apply(record);
+
+        Headers expected = new ConnectHeaders().addInt("AAA", 50000);
+
+        assertEquals(1, transformedRecord.headers().size());
+        assertEquals(expected, transformedRecord.headers());
+    }
+
+    @Test
+    public void insertHeaderWithInt64Value() {
+        final Map<String, Object> props = new HashMap<>();
+        props.put("header", "AAA");
+        props.put("literal.value", "87474836647");
+
+        xform.configure(props);
+
+        final SourceRecord record = new SourceRecord(null, null, "test",
+            0, null, null, null, null, null, null);
+        final SourceRecord transformedRecord = xform.apply(record);
+
+        Headers expected = new ConnectHeaders().addLong("AAA", 87474836647L);
+
+        assertEquals(1, transformedRecord.headers().size());
+        assertEquals(expected, transformedRecord.headers());
+    }
+
+    @Test
+    public void insertHeaderWithFloatValue() {
+        final Map<String, Object> props = new HashMap<>();
+        props.put("header", "AAA");
+        props.put("literal.value", "2353245343456.435435");
+
+        xform.configure(props);
+
+        final SourceRecord record = new SourceRecord(null, null, "test",
+            0, null, null, null, null, null, null);
+        final SourceRecord transformedRecord = xform.apply(record);
+
+        Headers expected = new ConnectHeaders().addDouble("AAA", 2353245343456.435435);
+
+        assertEquals(1, transformedRecord.headers().size());
+        assertEquals(expected, transformedRecord.headers());
+    }
+
+    @Test
+    public void insertHeaderWithDateValue() throws ParseException {
+        final Map<String, Object> props = new HashMap<>();
+        props.put("header", "AAA");
+        props.put("literal.value", "2019-08-23");
+
+        xform.configure(props);
+
+        final SourceRecord record = new SourceRecord(null, null, "test",
+            0, null, null, null, null, null, null);
+        final SourceRecord transformedRecord = xform.apply(record);
+
+        SchemaAndValue schemaAndValue = new SchemaAndValue(org.apache.kafka.connect.data.Date.SCHEMA,
+            new SimpleDateFormat("yyyy-MM-dd").parse("2019-08-23"));
+
+        Headers expected = new ConnectHeaders().add("AAA", schemaAndValue);
+
+        assertEquals(1, transformedRecord.headers().size());
+        assertEquals(expected, transformedRecord.headers());
+    }
+
+    @Test
+    public void insertHeaderWithTimeValue() throws ParseException {
+        final Map<String, Object> props = new HashMap<>();
+        props.put("header", "AAA");
+        props.put("literal.value", "14\\:34\\:54.346Z");
```

Comment:
Why are we escaping the colon character here? Why not the following?
```suggestion
        props.put("literal.value", "14:34:54.346Z");
```

It is true that `Properties.store()` will escape colons, but that shouldn't enter into how Kafka reads these property files. For example, `bootstrap.servers` property used in all of the Kafka Java client configuration files use `:` without escaping.

Reply:
I have detaield this in a previous comment:
The string gets tokenized in the method _Values.consumeNextToken because it contains the character ':', and is not treated like a whole. So I have to put the header value escaping the colons "14\\\\:34\\\\:22.222Z" and then delete the escape characters in the Values.parse method.

Just try to comment the line `token = token.replace("\\", "");` and put `props.put("literal.value", "14:34:54.346Z");` in the test `insertHeaderWithTimeValue`.
The test result is this:
```
java.lang.AssertionError: 
Expected :ConnectHeaders(headers=[ConnectHeader(key=AAA, value=Thu Jan 01 14:34:54 CET 1970, schema=Schema{org.apache.kafka.connect.data.Time:INT32})])
Actual   :ConnectHeaders(headers=[ConnectHeader(key=AAA, value=14, schema=Schema{INT8})])
```

Reply:
So that's a signal that there's a bug in the `Values.parseString(String)` method, so rather than fix it in this SMT it's better to fix it in the `Values` class. I've logged https://issues.apache.org/jira/browse/KAFKA-9074, and fixing that separately will make it much easier to backport the fix and will make this PR more straightforward.

Reply:
See https://github.com/apache/kafka/pull/7568 for my proposed fix.

Reply:
I completely agree with you, I also tried to modify the parseString method but I saw that I needed to be careful because it's a widely used method, so I wanted to hear your suggestions. Obviously your proposal is much more powerful!

Reply:
I hope to get #7568 approved and merged in the next few days. Once that's there, you can update this PR to remove that custom logic -- though I think we still could use tests that try to set the header value with a timestamp literal that does not escape the colons, and another test that escapes the colon. WDYT?

Reply:
I don't see the point in taking into account the case with escaped semicolons if you already can parse it without escaping them. I would remove those tests and the escaped semicolons replacement line in the parse method. So in this case 14\\\\:02\\\\:52 would not be an option.

Reply:
I guess I was suggesting having both escaped and unescaped versions just for completeness and to verify that both do indeed work without having to understand the `Values` implementation to know that escaped also works. But I will leave it up to you as to whether it's worth the extra test.

Reply:
Maybe you are right... It would be better to have both versions for completeness. As soon as your PR gets merged I will create the additional tests.